airflow/providers/google/cloud/hooks/dataproc.py

```diff
@@ -227,7 +237,10 @@ def get_job_client(self, region: Optional[str] = None) -> JobControllerClient:
             client_options = ClientOptions(api_endpoint=f'{region}-dataproc.googleapis.com:443')
 
         return JobControllerClient(
-            credentials=self._get_credentials(), client_info=CLIENT_INFO, client_options=client_options
+            credentials=self._get_credentials(),
+            client_info=CLIENT_INFO,
+            client_options=client_options,
+            transport="grpc_asyncio" if self.async_client else "grpc",
```

Comment:
In the Dataproc SDK we have `JobControllerAsyncClient` so maybe we should consider to use it: https://github.com/googleapis/python-dataproc/blob/main/google/cloud/dataproc_v1/services/job_controller/async_client.py#L43

Reply:
Thank you for the suggestion. Done